### PR TITLE
[EFI-875] chore: remove uneccessary usage of react-uid

### PR DESCRIPTION
**Pull request by jeremiah-clothier**
### Summary:

Removed usage of `useUID` from the three following components: `<Icon />`, `<PrimaryNav />` , and `<Breadcrumbs />`. Note, the `id` prop can still be passed to the components.

Why are we making this change?
To remove `console.error` messages from SSR in remix-run. Rendering these components would log errors like
> Warning: Prop `id` did not match. Server: "1" Client: "2"

### Test Plan:
- [x] Built library locally and validated the `console.error` message no longer appear when rendering these components

Follow-up, there is still a fair bit of usage of `react-uid` in the EDS repo. Those IDs are being used for accessibility and will be dealt with in a separate PR. This PR removes unnecessary usages

### CODE REVIEW

**Comment by booc0mtaco** [comment]
@jeremiah-clothier for the components where we remove react-uid, can you make sure we have a test that confirms `id` can be passed down?

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -148,7 +145,7 @@ export const Icon = (props: IconProps) => {
   if (purpose === 'informative') {
     return (
       <svg {...svgCommonProps} role="img">
-        <title id={idVar}>{props.title}</title>
+        <title id={id}>{props.title}</title>
```
Comment (ahuth) [inline src/components/Icon/Icon.tsx:148]: 👍

**Review by ahuth — APPROVED** [review]
LGTM 🤘 

Thanks for the cleanup

**Comment by jeremiah-clothier** [comment]
> @jeremiah-clothier for the components where we remove react-uid, can you make sure we have a test that confirms `id` can be passed down?

Does typescript + eslint count as a test? The `id` prop is destructured from props in each of the components which means if it wasn't being used then eslint would flag the unused variable and CI wouldn't pass

**Review by booc0mtaco — APPROVED** [review]

**Pull request merged**